src/Package/Impl/DataInspect/Viewers/CodeViewer.cs

```diff
@@ -2,28 +2,26 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.ComponentModel.Composition;
 using System.IO;
-using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.R.Components.Extensions;
 using Microsoft.R.Core.Formatting;
 using Microsoft.R.DataInspection;
 using Microsoft.R.Editor.Settings;
 using Microsoft.R.Host.Client;
 using Microsoft.R.Host.Client.Session;
-using Microsoft.VisualStudio.R.Package.Repl;
 using Microsoft.VisualStudio.R.Package.Shell;
-using static System.FormattableString;
 
 namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
     [Export(typeof(IObjectDetailsViewer))]
-    internal sealed class FunctionViewer : ViewerBase, IObjectDetailsViewer {
-        private readonly static string[] _classes = new string[] { "function", "formula" };
+    internal sealed class CodeViewer : ViewerBase, IObjectDetailsViewer {
+        private readonly static HashSet<string> _types = new HashSet<string> { "closure", "language" };
```

Comment:
I also don't see the value of this change - you don't have any unions or intersections, array has `Exists` and `Find` methods

---
In reply to: [63196108](https://github.com/Microsoft/RTVS/pull/1770#discussion_r63196108) [](ancestors = 63196108)